### ISSUE TITLE
Quality 

### ISSUE THREAD

**Issue by abylouw** [wetdog/wavenext_pytorch#1]
Hi,

Thank you for creating this repo and implementing the architecture in the paper. I have been looking at the paper and was going to try an implementation.

Do you have any preliminary results available? Do you think that it is better than for example iSTFTNet or MB-MelGan?

**Comment by wetdog** [comment]
I've not compared yet with iSTFTNet or MB-MelGan, but i'll try to hear those models on the same sample. Here is a sample of a first model. 
![image](https://github.com/wetdog/wavenext_pytorch/assets/8949105/f124d474-9d44-4b45-b9b1-f88b484e9b07)
 src: https://drive.google.com/file/d/1awn-oHt-wycZFyB7c_tQtC8hIewbrwR1/view?usp=drive_link
wavenext: https://drive.google.com/file/d/1jUDebB0oxuzo7VMu8pQEII3UXdtv0kLs/view?usp=drive_link

I'll probably try to increase the alpha of mrd loss to 1.0 as this was suggested here https://github.com/gemelo-ai/vocos/issues/48
I'll also post the weights of the trainings when it ends.

**Comment by egorsmkv** [comment]
@abylouw interesting to compare outputs for Ukrainian as well

I have a RAD-TTS model with these vocoders:

- https://github.com/egorsmkv/radtts-vocos
- https://github.com/egorsmkv/radtts-hifigan
- https://github.com/egorsmkv/radtts-istftnet
- https://github.com/egorsmkv/radtts-waveode

**Comment by patriotyk** [comment]
> I've not compared yet with iSTFTNet or MB-MelGan, but i'll try to hear those models on the same sample. Here is a sample of a first model. ![image](https://private-user-images.githubusercontent.com/8949105/325186958-f124d474-9d44-4b45-b9b1-f88b484e9b07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NTM1MjAsIm5iZiI6MTcxNDY1MzIyMCwicGF0aCI6Ii84OTQ5MTA1LzMyNTE4Njk1OC1mMTI0ZDQ3NC05ZDQ0LTRiNDUtYjliMS1mODhiNDg0ZTliMDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDJUMTIzMzQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZThiNGM2MTA2Y2M4MDUwYWQxNGMxNmM4ZWU2ZDlmN2Y4YjJkYzUwZDVkZWU3ZGQyZmYxYjM2YWMyODhiMzM2YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.19YUJMfn8zPzp4rPIyNRsdyoKvFFGkeCDs9ZHgeZj5o) src: https://drive.google.com/file/d/1awn-oHt-wycZFyB7c_tQtC8hIewbrwR1/view?usp=drive_link wavenext: https://drive.google.com/file/d/1jUDebB0oxuzo7VMu8pQEII3UXdtv0kLs/view?usp=drive_link
> 
> I'll probably try to increase the alpha of mrd loss to 1.0 as this was suggested here [gemelo-ai/vocos#48](https://github.com/gemelo-ai/vocos/issues/48) I'll also post the weights of the trainings when it ends.

I am training your vocos-matcha with  mrd loss = 1.0 and 44100 Hz So it is very slow, for almost 1M iterations, it still sounds slightly worse than yours https://huggingface.co/BSC-LT/vocos-mel-22khz. And metrics still worse.
The following logs is for about 3 weeks continuous training on 2 RTX3090 with dataset about 800 hours:
<img width="1069" alt="Знімок екрана 2024-05-02 о 15 55 22" src="https://github.com/wetdog/wavenext_pytorch/assets/1070523/9393def6-f138-4efd-8829-68781f886357">

**Comment by mush42** [comment]
Hi @wetdog 

What's the status of this implementation in terms of quality and speed?
Do you have pre-trained weights available?

I've great expectations for this repo 🙂
Best of luck!

**Comment by wetdog** [comment]
Hi @mush42 I finished a training for the mel version this week. in terms of quality it achieves better periodicity,
pesq_score, pitch_loss than vocos trained on the same datasets. you can find the weights here: https://huggingface.co/BSC-LT/wavenext-mel

![image](https://github.com/wetdog/wavenext_pytorch/assets/8949105/8f21fc5b-8c86-4003-b1ba-f5aa35af6782)

Also I fixed some things with the encodec experiment this week and now is training. For this trainings I used the mel features compatible with hifigan but probably is worth to train a version with 24khz using the same features as the original vocos. Let me know if you have some doubts.

**Comment by wetdog** [comment]
@egorsmkv Great work  I would probably use your versions to run some metrics and compare the quality of those vocoders.

**Comment by patriotyk** [comment]
@wetdog I have added your wavenext pretrained model to my huggingface app that runs pflowtts model. But unfortunately it sounds not very good. There is 4 vocoders that generate all waveforms from the same mel spectrogram generated by pflowtts and wavenext sounds similar to hifigan but slightly worse. There are also 44100 vocos vocoder trained from your implementation and it sounds the best. You can check it here https://huggingface.co/spaces/patriotyk/pflowtts_ukr_demo

**Comment by wetdog** [comment]
@patriotyk Thanks for the quick implementation, Do you think that this could be due to the dataset where it was trained? I used libritts for this run but I would like to try a version with commophone https://arxiv.org/abs/2201.05912 to make it more "universal".

**Comment by mush42** [comment]
Hi

Heavy TTS user here.
I don't agree with @patriotyk on this.
My initial testing shows that wavenext is significantly better than vocos, both in inference speed and synthesis quality.

Specifically, there is an audible hissing noise in the audio vocoded by vocos, probably as an ISTFT artifact.

Here's a sample of an unseen speaker, where Matcha TTS is used to generate the melspectogram.
[vocos-vs-wavenext.zip](https://github.com/user-attachments/files/15742963/vocos-vs-wavenext.zip)


Best
Musharraf

**Comment by patriotyk** [comment]
@wetdog I don't know, but seems to be yes. I will try pflowtts trained on libritts and we will see.

@mush42 Your Matcha TTS is trained on which dataset? Also your vocos sample is really bad, what pretrained model do you use here? On my app 'BSC-LT/vocos-mel-22khz' sounds much better.

**Comment by mush42** [comment]
@patriotyk 
Matcha was trained on [HifiCaptin](https://ast-astrec.nict.go.jp/en/release/hi-fi-captain/) US English female dataset.
I'm using an ONNX model converted from [this model](https://huggingface.co/BSC-LT/vocos-mel-22khz) with a custom ISTFT implementation that uses CNN (in order to be ONNX exportable).

**Comment by fd873630** [comment]
> Hi @mush42 I finished a training for the mel version this week. in terms of quality it achieves better periodicity, pesq_score, pitch_loss than vocos trained on the same datasets. you can find the weights here: https://huggingface.co/BSC-LT/wavenext-mel
> 
> ![image](https://private-user-images.githubusercontent.com/8949105/337641733-8f21fc5b-8c86-4003-b1ba-f5aa35af6782.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MTIxNTcsIm5iZiI6MTcxOTQxMTg1NywicGF0aCI6Ii84OTQ5MTA1LzMzNzY0MTczMy04ZjIxZmM1Yi04Yzg2LTQwMDMtYjFiYS1mNWFhMzVhZjY3ODIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjZUMTQyNDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmU0NTAwMDYyZDhjZThlYmJlZmRlMmU3ZGNlZGQ1NDc0OGM2YmYyNzhiNzAxOTBiZjc0MzA2NTY1ZDk2YTUxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9r7HHtWEIcgako9lZFXd1MsncVmAXZrn7J41hM2M59Q)
> 
> Also I fixed some things with the encodec experiment this week and now is training. For this trainings I used the mel features compatible with hifigan but probably is worth to train a version with 24khz using the same features as the original vocos. Let me know if you have some doubts.

Hi! @wetdog

Could you please share the .ckpt checkpoint file in addition to the .bin checkpoint file that you provided?

I want finetuning! but .bin checkpoint exist only generator!

**Comment by wetdog** [comment]
@fd873630 I just uploaded the ckpt. you can find it here https://huggingface.co/BSC-LT/wavenext-mel/blob/main/wavenext_2M_libritt_r.ckpt

**Comment by mush42** [comment]
@wetdog
Thanks for open-sourcing your work guys.
Really appreciate it.

Best
Musharraf

**Comment by polarathene** [comment]
> For this trainings I used the mel features compatible with hifigan but probably is worth to train a version with 24khz using the same features as the original vocos.

Did this ever happen?

I assume based on [this Vocos PR](https://github.com/gemelo-ai/vocos/pull/47#discussion_r2464186367) from you that "same features" also refers to `n_mels` in the config which was originally a value of 100, rather than 80 `n_mels` adapted (_which as the PR link notes was to get Vocos compatibility for processing outputs from acoustic models like Match-TTS that were trained with [mel spectrogram feature extraction from HiFi-GAN](https://github.com/search?q=repo%3Ajik876%2Fhifi-gan%2080&type=code)_)?

F5-TTS I think relies on the `n_mels: 100` 24KHz Vocos.
